# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/gcc/hotfix-glibc-2.42.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

../../../../libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp:489:31: error: invalid application of 'sizeof' to incomplete type '__sanitizer::termio'
  489 |   unsigned struct_termio_sz = sizeof(struct termio);
      |                               ^~~~~
#if SANITIZER_GLIBC || SANITIZER_ANDROID
  unsigned struct_termio_sz = sizeof(struct termio);
#endif

--- gcc-15.1.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp.vanilla	2025-07-29 13:09:27.591641995 +0200
+++ gcc-15.1.0/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp	2025-07-29 13:12:41.136946537 +0200
@@ -48,7 +48,7 @@
 #include <sys/times.h>
 #include <sys/types.h>
 #include <sys/utsname.h>
-#include <termios.h>
+#include <term.h>
 #include <time.h>
 #include <wchar.h>
 #include <regex.h>
@@ -486,7 +486,7 @@
   unsigned struct_mtpos_sz = sizeof(struct mtpos);
   unsigned struct_rtentry_sz = sizeof(struct rtentry);
 #if SANITIZER_GLIBC || SANITIZER_ANDROID
-  unsigned struct_termio_sz = sizeof(struct termio);
+  //unsigned struct_termio_sz = sizeof(struct termio);
 #endif
   unsigned struct_vt_consize_sz = sizeof(struct vt_consize);
   unsigned struct_vt_sizes_sz = sizeof(struct vt_sizes);
